### ISSUE TITLE
Support custom values for select

### ISSUE THREAD

**Issue by sirbris** [vaadin/vaadin-grid-pro#148]
Would be really helpful for the editor combobox (select component) to allow custom values like the ComboBox component itself

**Comment by yuriy-fix** [comment]
Hello!

That is already possible to achieve with [`Custom Edit Mode Renderer`](https://vaadin.com/components/vaadin-grid-pro/html-examples/grid-pro-edit-column-demos#custom-edit-mode-renderer) (column **Title (edit)**) by using `vaadin-combo-box` component as cell editor.

Column with `editor-type="select"` is using `vaadin-select` component internally which is a different component that doesn't support custom values.

**Comment by sirbris** [comment]
That's true, but styling is off in that case (row gets heigher, borders are shown, focus styling is set, margins etc.). It would require an extra css module to achieve the same vaadin-select look & feel inside the cell.

**Comment by rolfsmeds** [comment]
I created a separate ticket about creating a theme variant for text-field (and its derivatives like combo-box) that would remove most of the regular styling of the field for use cases like this: https://github.com/vaadin/vaadin-text-field/issues/546.

In the meantime, creating a custom css module for this should be fairly straightforward by copying the css from the module grid-pro uses for its internal version of text-field: https://github.com/vaadin/vaadin-grid-pro/blob/master/theme/lumo/vaadin-grid-pro-edit-text-field.html
